quizgen/latex.py

```diff
@@ -3,32 +3,81 @@
 import shutil
 import subprocess
 
+import quizgen.util.dirent
+
 _pdflatex_bin_path = None
+_pdflatex_use_docker = False
+
+DOCKER_IMAGE = "quizgen/latex.py"
 
 def set_pdflatex_bin_path(path):
     global _pdflatex_bin_path
     _pdflatex_bin_path = path
 
-def is_available():
-    if (_pdflatex_bin_path is not None):
-        return True
-
-    if (shutil.which('pdflatex') is None):
-        logging.warning("Could not find `pdxlatex`, cannot compile PDFs")
-        return False
+def set_pdflatex_use_docker(use_docker):
+    global _pdflatex_use_docker
+    _pdflatex_use_docker = use_docker
 
-    return True
+def is_available(use_docker = False):
+    if use_docker:
+        result = subprocess.call(["docker", "--version"], stdout = subprocess.DEVNULL, stderr = subprocess.DEVNULL)
+        return result == 0
+    return _pdflatex_bin_path is not None or shutil.which('pdflatex') is not None
 
 def compile(path):
-    bin_path = 'pdflatex'
-    if (_pdflatex_bin_path is not None):
-        bin_path = _pdflatex_bin_path
+    if _pdflatex_use_docker:
+        return _compile_docker(path)
+    return _compile_local(path)
+
+def _compile_local(path):
+    bin_path = _pdflatex_bin_path or "pdflatex"
+    result = subprocess.run(
+        [bin_path, "-interaction=nonstopmode", os.path.basename(path)],
+        cwd=os.path.dirname(path),
+        capture_output = True
+    )
+    return result.returncode == 0
+
+def _compile_docker(path):
+    temp_dir = quizgen.util.dirent.get_temp_path()
+    temp_tex = os.path.join(temp_dir, os.path.basename(path))
+    
+    quizgen.util.dirent.copy_dirent(path, temp_tex)
+    
+    source_dir = os.path.dirname(path)
+    images_dir = os.path.join(source_dir, 'images')
+    if os.path.exists(images_dir):
+        dest_images_dir = os.path.join(temp_dir, 'images')
+        shutil.copytree(images_dir, dest_images_dir, dirs_exist_ok = True)
```

Comment:
I don't like the special handling of the `images` directory.
I especially don't like hardcoding the name "images".

Instead, why not pass in a list of additional paths that are required (default to and empty list).
Note that this will require a comment describing the semantics (e.g., all extra paths will be placed adjacent to the tex file).